general/front-end-configs.md

```diff
@@ -0,0 +1,183 @@
+```js
+module.exports = {
+  extends: '@avenga/stylelint-config',
+};
+```
+
+If you need to further customise the ruleset for your project you can look into [the official docs](https://stylelint.io/user-guide/configuration/#the-configuration-object).
+
+### Editor Plugins
+
+- [Atom](https://atom.io/packages/linter-stylelint)
+- [vim](https://github.com/vim-syntastic/syntastic/blob/master/syntax_checkers/css/stylelint.vim)
+- [Visual Studio Code](https://github.com/stylelint/vscode-stylelint)
+- [Sublime Text](https://github.com/SublimeLinter/SublimeLinter-stylelint)
+- [WebStorm](https://www.jetbrains.com/help/webstorm/using-stylelint-code-quality-tool.html)
+
+## ESLint
+
+[ESLint](https://eslint.org/) has been the standard for lining JavaScript-based code for quite a while. It enjoys a solid standing throughout the whole Open-Source Community and is being known for it's rock solid support.
+
+### Usage
+
+Installing eslint:
+
+```bash
+npm install --save-dev eslint @avenga/eslint-config
+```
+
+Add the configuration file `.eslintrc` with the following content:
+
+```json
+{
+  "extends": "@avenga/eslint-config"
+}
+```
+
+If you need to further customise the ruleset for your project you can look [into the official docs](https://eslint.org/docs/user-guide/configuring).
+
+### Editor Plugins
+
+- [Atom](https://atom.io/packages/linter-eslint)
+- [vim](https://github.com/vim-syntastic/syntastic/tree/master/syntax_checkers/javascript)
+- [Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
+- [Sublime Text](https://github.com/SublimeLinter/SublimeLinter-eslint)
+- [WebStorm](https://www.jetbrains.com/help/webstorm/eslint.html)
+
+## Git Hooks
+
+Git hooks can be used to execute arbitrary scripts or commands on git events. The most popular one is the `pre-commit` hook which runs just before a git commit is created. We recommend create a `pre-commit` hook for [prettier](#prettier) so that only formatted code can enter the repository.
+
+If there isn't already a git hook pipeline setup in your project we suggest setting them up via [husky](https://github.com/typicode/husky) and [lint-staged](https://github.com/okonet/lint-staged). _Note that for backend projects that are not based on nodejs you'll likely encounter different tools for setting up git hooks._
+
+```bash
+npm install --save-dev husky
+```
+
+Husky is used to execute npm scripts defined in `package.json` as a git hook.
+
+```json
+{
+  "name": "my-project",
+  "husky": {
+    "hooks": {
+      "pre-commit": "echo 'hello world'"
+    }
+  },
+  "devDependencies": {
+    "husky": "^x.x.x"
+  }
+}
+```
+
+This will output `hello world` before creating a commit.
+
+If you execute commands that expect files as input, like it is commonly done with linters, you do not want to lint the whole repository. Instead, developers only want to check the files that are staged in Git. This is a lot more performant and ensures a less invasive workflow change.
+
```

Comment:
Add: 'P.S. Remember to provide proper husky version.'

Reply:
this is mentioned in the devDependencies, isn't it?

Reply:
see line 150 it's just
```
"devDependencies": {
    "husky": "^x.x.x"
  }
```